posthog/api/test/notebooks/test_notebook_filtering.py

```diff
@@ -1,74 +1,280 @@
-from typing import Dict, Any
+from typing import Dict, Any, List
 
 from rest_framework import status
 
 from posthog.test.base import APIBaseTest, QueryMatchingTest
 
+PLAYLIST_CONTENT = lambda: {
+    "type": "ph-recording-playlist",
+    "attrs": {
+        "filters": '{"specific_filters": "in here"}',
+    },
+}
+
+FEATURE_FLAG_CONTENT = lambda id: {
+    "type": "ph-feature-flag",
+    "attrs": {
+        "id": id or "feature_flag_id",
+    },
+}
+
+PERSON_CONTENT = lambda id: {"type": "ph-person", "attrs": {"id": id or "person_id"}}
+
+RECORDING_CONTENT = lambda id: {"type": "ph-recording", "attrs": {"id": id or "session_recording_id"}}
+RECORDING_COMMENT_CONTENT = lambda id: {
+    "type": "ph-replay-timestamp",
+    "attrs": {"playbackTime": 52000, "sessionRecordingId": id or "session_recording_id"},
+}
+
+INSIGHT_COMMENT = lambda id: {
+    "type": "ph-insight",
+    "attrs": {
+        "id": id or "insight_short_id",
+    },
+}
+
 
 class TestNotebooksFiltering(APIBaseTest, QueryMatchingTest):
-    def _create_notebook_with_content(self, content_with_a_recording: Dict[str, Any]) -> str:
+    def _create_notebook_with_content(self, inner_content: List[Dict[str, Any]]) -> str:
         response = self.client.post(
-            f"/api/projects/{self.team.id}/notebooks", data={"content": content_with_a_recording}
+            f"/api/projects/{self.team.id}/notebooks",
+            data={
+                "content": {
+                    "type": "doc",
+                    "content": inner_content,
+                }
+            },
         )
         assert response.status_code == status.HTTP_201_CREATED
         return response.json()["id"]
 
-    def test_notebook_with_any_recording(self) -> None:
-        content_with_a_recording = {
-            "type": "doc",
-            "content": [
-                {
-                    "type": "ph-recording",
-                }
-            ],
-        }
-
-        content_without_recording = {
-            "type": "doc",
-            "content": [
-                {
-                    "type": "paragraph",
-                }
-            ],
-        }
+    def test_filtering_by_types(self) -> None:
```

Comment:
these are pretty chunky tests but I thought the parameterized version would be too complex and individual tests way more verbose

Reply:
This _is_ a chunky test 🙈